<?xml version="1.0"?>
<launch>
  <arg name="paused" default="false"/>
  <arg name="gazebo_gui" default="true"/>
  <arg name="start_world" default="false"/>

  <!-- startup simulated world -->
  <include if="$(arg start_world)" file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" default="worlds/empty.world"/>
    <arg name="verbose" value="true"/>
    <arg name="paused" value="$(arg paused)"/>
    <arg name="gui" value="$(arg gazebo_gui)"/>
  </include>

  <!-- send robot urdf to param server -->
  <param name="robot_description" command="$(find xacro)/xacro '$(find franka_description)/robots/panda_arm_hand.urdf.xacro' use_cylinder_collision_model:=false use_gazebo_sim:=true"/>

  <!-- push robot_description to factory and spawn robot in gazebo at the origin, change x,y,z arguments to spawn in a different position -->
  <node name="spawn_gazebo_model" pkg="gazebo_ros" type="spawn_model" args="-urdf -param robot_description -model panda -x 0 -y 0 -z 0
              -J panda_joint4 -0.07 -J panda_joint6 0.51 -unpause" respawn="false" output="screen" />

  <include file="$(find panda_moveit_config)/launch/ros_controllers.launch"/>

</launch>
